pkg/controller/node.go

```diff
@@ -96,6 +98,36 @@ func (rm *ResourceManager) DrainPods(nodeName string, unhealthyAfter time.Durati
 	return false, k8s.DrainPods(pods, rm.RawClient, unhealthyAfter)
 }
 
+func (rm *ResourceManager) AddNodegroupAnnotationToNode(nodeName, nodegroupName string) error {
```

Comment:
Is there a reason for using an annotation for the node group rather than a label?

Reply:
Mainly because it's not identifying the CNR itself but rather is used by the controller to link back to the ASG.

Reply:
I guess more question was more around - we already have logic for adding a label to a node in the Cyclops codebase - can we re-use it?

Reply:
Yeah there's nothing stopping us from doing that, if that's preferable happy to go that way. I did it because it was minimal work and more strictly adheres to the definitions from what I understand.